components/pylontech.rst

```diff
@@ -0,0 +1,229 @@
+Pylontech Battery
+=================
+
+.. seo::
+    :description: Instructions for setting up Pylontech Battery in ESPHome.
+    :image: pylontech.jpg
+
+The Pylontech component allows you to pull data from Pylontech Batteries into ESPHome.
+It uses :ref:`UART <uart>` for communication.
+
+Once configured, you can use sensors as described below for your projects.
+
+.. figure:: images/pylontech.jpg
+    :align: center
+    :width: 50.0%
+
+    Pylontech Batteries
+
+Hardware Setup
+--------------
+
+You can connect to Pylontech Batteries using the port labeled "Console".
+Any connections via CAN or RS485 (e.g. to an inverter) are untouched and remain functional.
+
+The console port offers a RS232 interface using a RJ10 or RJ45 connector.
+The voltage levels are *not* TTL-compatible. A RS232 transceiver must be placed between the Batteries and the ESPHome device.
+MAX3232-based transceivers have been tested and work well.
+
+If you have multiple batteries you need to connect to the master battery's console port.
+
+.. list-table:: Pylontech RJ45 Console Port (US2000C, US3000C)
+    :header-rows: 1
+
+    * - RJ45 Pin
+      - Function
+      - Connect to
+    * - 3
+      - Pylontech TX
+      - ESPHome RX via transceiver
+    * - 6
+      - Pylontech RX
+      - ESPHome TX via transceiver
+    * - 8
+      - GND
+      - GND
+
+.. figure:: images/rj45_pinout.jpg
+    :align: center
+    :width: 70.0%
+
+    RJ45 pinout
+
+.. list-table:: Pylontech RJ10 Console Pinout (US2000B)
+    :header-rows: 1
+
+    * - RJ410 Pin
+      - Function
+      - Connect to
+    * - 2
+      - Pylontech RX
+      - ESPHome TX via transceiver
+    * - 3
+      - Pylontech TX
+      - ESPHome RX via transceiver
+    * - 4
+      - GND
+      - GND
+      
+Configuration
+-------------
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      - id: pylontech_uart
+        tx_pin: GPIO1
+        rx_pin: GPIO3
+        baud_rate: 115200
+
```

Comment:
We don't add external examples here. It is up to the user to visit the uart docs to see how to configure that for their use case.
```suggestion
    # Example configuration entry
```